artifacts/mainnet/ics/assessment/experience/main.py

```diff
@@ -0,0 +1,241 @@
+
+
+def sdvtm_score(addresses: Iterable[str]) -> int:
+    """
+    Returns the score for SDVTM participation if any address is eligible, otherwise 0.
+    """
+    if is_addresses_in_csv(addresses, "sdvtm-mainnet.csv"):
+        return scores["sdvtm-mainnet"]
+    if is_addresses_in_csv(addresses, "sdvtm-testnet.csv"):
+        return scores["sdvtm-testnet"]
+    return 0
+
+
+def csm_score(addresses: Iterable[str]) -> int:
+    """
+    Returns the score for CSM participation if any address is eligible, otherwise 0.
+    This function checks both testnet and mainnet CSM participation.
+    """
+    testnet_score = _csm_testnet_score(addresses)
+    mainnet_score = _csm_mainnet_score(addresses)
+
+    if mainnet_score:
+        return mainnet_score
+    if testnet_score:
+        return testnet_score
+    return 0
+
+def _csm_testnet_score(addresses: Iterable[str]) -> int:
+    """
+    Returns the score for CSM testnet participation if any address is eligible, otherwise 0.
+    """
+    perf_reports = [
+        "QmTpTekd8qV9mn46pYzT9fkHtYHyQguZrbGdF233YYibvY"
+    ]
+    if _check_csm_performance_logs(
+            addresses,
+            "node_operator_owners_hoodi.json",
+            perf_reports,
+            "Testnet"  # Network name for logging
+    ):
+        return scores["csm-testnet"]
+    return 0
+
+def _csm_mainnet_score(addresses: Iterable[str]) -> int:
+    """
+    Returns the score for CSM mainnet participation if any address is eligible, otherwise 0.
+    """
+    perf_reports = [
+        "QmaHU6Ah99Yk6kQVtSrN4inxqqYoU6epZ5UKyDvwdYUKAS"
+    ]
+    if _check_csm_performance_logs(
+            addresses,
+            "node_operator_owners_mainnet.json",
+            perf_reports,
+            "Mainnet"  # Network name for logging
+    ):
+        return scores["csm-mainnet"]
+    return 0
+
+def _request_performance_report(report_file, retries=3, delay=2):
+    url = f"https://ipfs.io/ipfs/{report_file}"
+    for attempt in range(retries):
+        try:
+            response = requests.get(url)
+            response.raise_for_status()
+            return response.json()
+        except (requests.HTTPError, requests.JSONDecodeError) as e:
+            print(f"Error fetching report {report_file}: {e}")
+            if attempt < retries - 1:
+                time.sleep(delay)
+                continue
+            raise e
+    raise Exception(f"Failed to fetch report {report_file}")
+
+
+def _check_csm_performance_logs(addresses: Iterable[str], no_owners_file_name, perf_reports, network_name) -> bool:
+    """
+    Returns True if any address is a node operator with all validators above the threshold in all logs.
+    Used for both testnet and mainnet CSM checks.
+    """
+    with open(current_dir / no_owners_file_name, 'r') as f:
+        node_operators = json.load(f)
+
+    address_to_id = {}
+    for no_id, addr in node_operators.items():
+        address_to_id[addr.lower()] = no_id
+
+    addresses = set(addresses)
```

Comment:
Do I need to check at args validation stage?

Reply:
this is a set already, good one